source/statements.tex

```diff
@@ -1108,5 +1107,6 @@
                                 // since it depends on \tcode{T2} being a type-name
 }
 \end{codeblock}
-\end{example}%
+\end{example}
+%
```

Comment:
This looks broken: if there really was need to avoid a whitespace, then this newline introduces one.

My suspicion is that this wasn't necessary in the first place (in which case we should remove the `%`), but as far as this "cleanup change" goes, this seems undesirable.

Could you take a look if a) this happens anywhere else, and b) if the `%` was necessary in the first place?

Reply:
Yes, this happens a few times. Frequently at the end of a section, just before the closing index marker. So, the % is totally bogus and should / will be removed.

Reply:
Could you maybe send a zero-visual-diff commit first that removes the bogus `%` everywhere?

Reply:
Uh, that will produce conflicts of all those removals with the rest of the patch.

I've just browsed through all pages with visual ("Appearance") diffs and there's nothing bad going on, except for the well-documented "space before _-- end note]_ becomes more consistent".

Reply:
Ok, I've done an isolated patch that removes trailing % from `note}` and `example}`.

diffpdf in "Appearance" mode shows a single difference in 13.9.2.1, with additional space after an "-- end example}". Which actually looks better now, in my view.

I think the "%" removal is safe.